res/js/index.js

```diff
@@ -346,12 +348,51 @@ function handleGeolocationError(error) {
 	statusPos.html("Geolocation unavailable.");
 }
 
+var map = null;
+var markers = null;
+
+function updatePosMarker(e) {
+ 	if( map == null ) {
+		map = new OpenLayers.Map("mapdiv");
+		map.events.register('zoomend', map, storeMapZoomLevel);
+		map.addLayer(new OpenLayers.Layer.OSM());
+		markers = new OpenLayers.Layer.Markers( "Markers" );
+		map.addLayer(markers);
+	} else {
+		markers.clearMarkers();
+	}
+
+	var lat = $('#pos_lat').val();
+	var lon = $('#pos_long').val();
+	var lonLat = new OpenLayers.LonLat(lon, lat).transform(
+            new OpenLayers.Projection("EPSG:4326"), // transform from WGS 1984
+            map.getProjectionObject() // to Spherical Mercator Projection
+	);
+
+	var zoom=getMapZoomLevel();
+	map.setCenter(lonLat, zoom);
+	markers.addMarker(new OpenLayers.Marker(lonLat));
+}
+
+function storeMapZoomLevel() {
+	Cookies.set("ol-zoom", map.getZoom());
+}
+
+function getMapZoomLevel() {
+	var zoom = Cookies.get("ol-zoom");
+	if( zoom == null ){
+		zoom = 5;
+	}
+	return zoom;
+}
+
 function updatePositionGeolocation(pos) {
 	var d = new Date(pos.timestamp);
 	statusPos.html("Last position update " + dateFormat(d) + "...");
 	$('#pos_lat').val(pos.coords.latitude);
 	$('#pos_long').val(pos.coords.longitude);
 	$('#pos_ts').val(pos.timestamp);
+	updatePosMarker(null);
```

Comment:
Instead of calling `updatePosMarker(null)` after each position update, you could consider the ajax complete callback :
```
complete: function( jqXHR, textStatus ){
	updatePosMarker(null);
}
```
https://github.com/la5nta/pat/blob/7bf89d2beb5700111bcff3c5227dc47af0d6dfa4/res/js/index.js#L54-L77

Otherwise the map div does not show the map at all (if no manual imput, GPSd and Geolocation unavailable).